src/System.ComponentModel.Annotations/tests/DataTypeAttributeTests.cs

```diff
@@ -68,6 +68,16 @@ public static void GetDataTypeName_and_IsValid_on_empty_custom_DataTypeAttribute
         }
 
         [Fact]
+        public static void GetDataTypeName_and_IsValid_on_whitespace_custom_DataTypeAttribute_throws_exception()
+        {
+            var attribute = new DataTypeAttribute(new String(' ', 5)); // Create a string with five whitespaces
```

Comment:
Adding more complicated tests sounds reasonable. In that case we should have both -- simple & complicated. You can use InlineData attribute for that (see [examples](https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=InlineData&type=Code)).